packages/atomic/src/components/result-template-components/atomic-result-badge/atomic-result-badge.tsx

```diff
@@ -47,16 +48,18 @@ export class AtomicResultBadge {
     );
   }
 
+  private getTextContent() {
+    if (this.field !== undefined) {
+      return <atomic-result-text field={this.field}></atomic-result-text>;
+    }
+    if (this.label !== undefined) {
+      return <atomic-text value={this.label}></atomic-text>;
+    }
+    return <slot></slot>;
```

Comment:
Should `<slot>` always be rendered, even if e.g. a `label` is specified? I feel this is the more natural behavior.

Also, should all components support a slot? Could this PR open the door to questions like "why am I able to slot here but not there"?

Reply:
> Should `<slot>` always be rendered, even if e.g. a `label` is specified? I feel this is the more natural behavior?

Agreed

> Also, should all components support a slot? Could this PR open the door to questions like "why am I able to slot here but not there"?

Good point. However, I'm not sure if there's other components where it makes as much sense to add a slot to as badges, since badges are a very small and purely visual component. In a way, from an implementer perspective, badges wouldn't be so different from result sections.